src/linter/root.go

```diff
@@ -568,14 +569,19 @@ func (d *RootWalker) handleComment(c freefloating.String) {
 		return
 	}
 
+	if d.handledFiles == nil {
+		d.handledFiles = make(map[string]bool)
```

Comment:
As @YuriyNasretdinov already mentioned, you don't need a map here. Root walker is file-scoped. It doesn't handle multiple files. Having a bool field should be enough.